pkg/aspect/clean/darwin.go

```diff
@@ -0,0 +1,46 @@
+//go:build darwin
+
+package clean
+
+import (
+	"io/fs"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+)
+
+func (c *Clean) GetAccessTime(workspace fs.FileInfo) time.Duration {
+	accessTime := workspace.Sys().(*syscall.Stat_t).Atimespec
+	createdTime := workspace.Sys().(*syscall.Stat_t).Ctimespec
+	modifiedTime := workspace.Sys().(*syscall.Stat_t).Mtimespec
+
+	timeSinceAccess := time.Since(time.Unix(accessTime.Sec, accessTime.Nsec))
+	timeSinceCreation := time.Since(time.Unix(createdTime.Sec, createdTime.Nsec))
+	timeSinceModified := time.Since(time.Unix(modifiedTime.Sec, modifiedTime.Nsec))
+
+	smallestTime := timeSinceAccess
+
+	if timeSinceCreation < timeSinceAccess && timeSinceCreation < timeSinceModified {
+		smallestTime = timeSinceCreation
+	} else if timeSinceModified < timeSinceAccess && timeSinceModified < timeSinceCreation {
+		smallestTime = timeSinceModified
+	}
+
+	return smallestTime
+}
+
+func (c *Clean) MoveFolderToTmp(dir string, name string) string {
+	newFolder := "/private/var/tmp/aspect_delete/" + strings.Replace(dir, "/", "", -1)
+	newPath := newFolder + "/" + name
+	os.MkdirAll(newFolder, os.ModePerm)
+	os.Rename(dir+"/external/"+name, newPath)
+
+	return newPath
+}
+
+func (c *Clean) ChangeFolderPermissions(folder string) ([]byte, error) {
+	cmd := exec.Command("chmod", "-R", "777", folder)
```

Comment:
use the `os` package rather than spawn a subprocess, for better portability https://pkg.go.dev/os#Chmod

Reply:
My reasoning for doing it this way was that there is no recursive mode for https://pkg.go.dev/os#Chmod so I will need to walk the file tree in order to change everything to have deletable permissions. But I can definitely switch. Thats not a problem. Thoughts?

Reply:
Since this is a specialized code for macOS, I'm fine with the subprocess. The `chmod` command is already optimized enough and with lots of baggage for doing this job correctly.